src/translate/constraints.py

```diff
@@ -1,41 +1,31 @@
 import itertools
+from typing import Iterable, List, Tuple
 
-class NegativeClause:
+class InequalityDisjunction:
     # disjunction of inequalities
-    def __init__(self, parts):
+    def __init__(self, parts: List[Tuple[str, str]]):
         self.parts = parts
         assert len(parts)
 
     def __str__(self):
-        disj = " or ".join(["(%s != %s)" % (v1, v2)
-                            for (v1, v2) in self.parts])
-        return "(%s)" % disj
-
-    def is_satisfiable(self):
-        for part in self.parts:
-            if part[0] != part[1]:
-                return True
-        return False
-
-    def apply_mapping(self, m):
-        new_parts = [(m.get(v1, v1), m.get(v2, v2)) for (v1, v2) in self.parts]
-        return NegativeClause(new_parts)
+        disj = " or ".join([f"({v1} != {v2})" for (v1, v2) in self.parts])
+        return f"({disj})"
 
 
-class Assignment:
+class EqualityConjunction:
     def __init__(self, equalities):
         self.equalities = tuple(equalities)
```

Comment:
It actually does not make a difference, it is just included for also permitting iterators as `equalities` argument and a tuple was more natural than a list (because it is not meant to be modified after construction). There actually is only one position in the code where `equalities` is not already a list, so we could change this there and just store the argument in the member variable. Would you find this preferable? (It might also save a litte bit of time.)

Reply:
I just found it confusing because it comes directly after InequalityDisjunction and since the two store very similar information I was wondering what the purpose of the difference is. If it can be made more similar without much effort or performance loss I'm all for it.

Reply:
I now always use a fresh list as an input and don't copy it in the initialization.